### PR TITLE
Add new TAP formatter

### PR DESCRIPTION
**Pull request by cyberdelia**
Add new TAP formatter, it implements the [Test Anything Protocol](https://testanything.org).

-----------------

Before submitting the PR make sure the following are checked:

* [x] Wrote [good commit messages][1].
* [x] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [x] Used the same coding conventions as the rest of the project.
* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [x] Squashed related commits together.
* [x] Added tests.
* [x] Added an entry to the [Changelog](../blob/master/CHANGELOG.md) if the new code introduces user-observable changes. See [changelog entry format](../blob/master/CONTRIBUTING.md#changelog-entry-format).
* [x] All tests are passing.
* [x] The new code doesn't generate RuboCop offenses.
* [x] The PR relates to *only* one subject with a clear title
  and description in grammatically correct, complete sentences.
* [x] Updated cop documentation with `rake generate_cops_documentation` (required only when you've added a new cop or changed the configuration/documentation of an existing cop).

[1]: http://chris.beams.io/posts/git-commit/

### CODE REVIEW

File: lib/rubocop/formatter/tap_formatter.rb [hunk 1]
```diff
@@ -0,0 +1,70 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Formatter
+    # This formatter formats report data in the TAP format.
```
Comment (bbatsov) [inline lib/rubocop/formatter/tap_formatter.rb]: I'd extend the description here a bit for the sake of people like me, who had never heard of this before. :-)

**Comment by bbatsov** [comment]
You should also extend the section of the documentation about formatters.

**Comment by cyberdelia** [comment]
@bbatsov I added a few words, and added the missing ``--format tap`` option.

**Comment by bbatsov** [comment]
👍

**Pull request merged**